### PR TITLE
Cleaned up MediaPickerActivity

### LINKED ISSUE #18: Remove `WPAndroid` and `FluxC` dependencies from `MediaPickerActivity`
Acceptance criteria: 
- [ ] searching for `org.wordpress.android.ui` in `MediaPickerActivity` returns 0 results.
- [ ] searching for `org.wordpress.android.util` in `MediaPickerActivity` returns 0 results.
- [ ] searching for `fluxc` in `MediaPickerActivity` returns 0 results.

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #18 by removing FluxC, Dagger and `WPAndroid` dependencies from `MediaPickerActivity`. 

### Changes
- Fixed build errors in `FileThumbnailViewHolder` and `LoaderViewHolder`
- Removed unused `MediaPickerActionModeCallback`. This is used by WP to provide editing option when using the media picker. 
- Removed `MediaModel` reference from `MediaNavigationEvent`.
- Removed reference to [LocaleAwareActivity](https://github.com/wordpress-mobile/WordPress-Android/blob/4585d19679b9f651d39dfd3372de544a093b221c/WordPress/src/main/java/org/wordpress/android/ui/LocaleAwareActivity.kt) - This is used by the WordPress app to ensure that locale changes are supported at an application level. I don't know if this is something we want to retain in this library so opened [an issue](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/issues/32) to keep track.
- Removed reference to `SiteModel` & `Dispatcher` from `MediaPickerActivity`.
- Added `MediaPickerConstants` and `MediaPickerRequestCodes` used in `MediaPickerActivity`.
- Added dependency for `Viewbinding` and `Kotlin parcelize`.
- Updated `WPMediaUtils` to include methods needed in `MediaPickerActivity`.
- Added 1 icon needed for `MediaPickerActivity`.

With these changes, the refactor to `MediaPickerActivity` is completed.

### CODE REVIEW

File: MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerRequestCodes.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.wordpress.android.mediapicker
+
+object MediaPickerRequestCodes {
+    const val IMAGE_EDITOR_EDIT_IMAGE = 6000
+
+    // Media
+    const val TAKE_PHOTO = 2100
```
Comment (wzieba) [inline MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerRequestCodes.kt:7]: 📌  Note to myself: fix this in https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/25/files#diff-9ce9ee2216456a5308e7ccc31091deaf9a9a49bdb41fb6f58539a19f975f6a0eR68-R69

File: MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerActionModeCallback.kt [hunk 2]
```diff
@@ -1,102 +0,0 @@
...
-import androidx.lifecycle.Lifecycle.Event.ON_STOP
-import androidx.lifecycle.LifecycleOwner
-import androidx.lifecycle.LifecycleRegistry
-import androidx.lifecycle.Observer
-import org.wordpress.android.R
-import org.wordpress.android.mediapicker.MediaPickerViewModel.ActionModeUiModel
-import org.wordpress.android.ui.utils.UiString.UiStringRes
-import org.wordpress.android.ui.utils.UiString.UiStringText
-
-class MediaPickerActionModeCallback(private val viewModel: MediaPickerViewModel) : Callback,
```
Comment (wzieba) [inline MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerActionModeCallback.kt:21]: 🔧  I think this class is used in `MediaPickerFragment` : https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/33/files#diff-7f84ecd9a646ff8eb7edd2717f7a4d5b4347da3be0ec1851d321b1573a25b9ebR253

Maybe we shouldn't remove it, WDYT?

File: MediaPicker/build.gradle [hunk 3]
```diff
@@ -1,6 +1,7 @@
 plugins {
     id 'com.android.library'
     id 'org.jetbrains.kotlin.android'
+    id 'org.jetbrains.kotlin.plugin.parcelize' version '1.5.30-M1'
```
Comment (wzieba) [inline MediaPicker/build.gradle]: 🔧  By using Gradle's Plugin DSL we now should rather set up versions of our plugins in `settings.gradle`. I've pushed a commit on separate branch: 7409305d5cd8f2bde0b3345c35b062f4c6233c6a - if you'd like to add this change to the PR, please cherry-pick the commit to this PR.

This is similar, modern setup [we have in FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/settings.gradle).

**Review by wzieba — COMMENTED** [review]
Thanks @anitaa1990 for working on this! I've left two comments, WDYT about them?

File: MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerActionModeCallback.kt (hunk 2 above)
Comment (anitaa1990) [reply MediaPicker/src/main/java/org/wordpress/android/mediapicker/MediaPickerActionModeCallback.kt:21]: That's a good point! Android studio didn't really show any usages so I just removed it after looking at the class and the fact that it focuses on menu options like editing, which we don't support at the moment. 

But I'll add it back to this PR and if I find it's not needed in `MediaPickerFragment` I'll remove it again.

File: MediaPicker/build.gradle (hunk 3 above)
Comment (anitaa1990) [reply MediaPicker/build.gradle]: TIL! Thank you :) I fixed this in 8ce600f following your advise

**Comment by anitaa1990** [comment]
Thank you for the review @wzieba! I fixed both those issues mentioned so ready for another round 👍

**Review by wzieba — APPROVED** [review]
Thank you @anitaa1990 looks good! :shipit:

**Pull request merged**